lsp/nls/src/requests/completion.rs

```diff
@@ -780,6 +796,80 @@ pub fn handle_completion(
     Ok(())
 }
 
+fn handle_import_completion(
+    import: &OsString,
+    params: &CompletionParams,
+    server: &mut Server,
+) -> io::Result<Vec<CompletionItem>> {
+    debug!("handle import completion");
+
+    let current_file = params
+        .text_document_position
+        .text_document
+        .uri
+        .to_file_path()
+        .unwrap();
+    let current_file = cache::normalize_path(current_file)?;
+
+    let mut current_path = current_file.clone();
+    current_path.pop();
+    current_path.push(import);
```

Comment:
Did you try not adding `import` to the path? If I understood correctly, your intention is that if they type `import "foo"` then you'll only try completing filenames starting with "foo"?

Because my understanding for how completion works in LSP is you're not supposed to filter the responses based on prefix: just return all the files in the right directory, and let the editor sort it out. Because the editor might want to filter by fuzzy prefix or something...